### PR TITLE
Make MP use KFS parameters as Timestamps

### PR DESCRIPTION
**Pull request by atobiszei**
JIRA:CVS-123559

### CODE REVIEW

File: src/mediapipe_internal/mediapipegraphexecutor.cpp [hunk 1]
```diff
@@ -442,10 +442,15 @@ enum : unsigned int {
...
 
 constexpr size_t STARTING_TIMESTAMP = 0;
+const std::string MediapipeGraphExecutor::TIMESTAMP_PARAMETER_NAME = "OVMS_MP_TIMESTAMP";
 
 static std::map<std::string, mediapipe::Packet> createInputSidePackets(const KFSRequest* request) {
     std::map<std::string, mediapipe::Packet> inputSidePackets;
     for (const auto& [name, valueChoice] : request->parameters()) {
+        SPDLOG_DEBUG("Found: {}; parameter in request for: {};", name, request->model_name());
+        if (name == MediapipeGraphExecutor::TIMESTAMP_PARAMETER_NAME) {
+            continue;
```
Comment (dkalinowski) [inline src/mediapipe_internal/mediapipegraphexecutor.cpp:452]: We could inform that we ignored timestamp parameter from deserialization

File: src/mediapipe_internal/mediapipegraphexecutor.cpp [hunk 2]
```diff
@@ -758,9 +763,8 @@ Status MediapipeGraphExecutor::infer(const KFSRequest* requestPtr, KFSResponse*
         }
         outputPollers.emplace(name, std::move(absStatusOrPoller).value());
     }
-
-    std::map<std::string, mediapipe::Packet> inputSidePackets{createInputSidePackets(requestPtr)};
-    absStatus = graph.StartRun(inputSidePackets);
+    std::map<std::string, mediapipe::Packet> sideInputPackets{createInputSidePackets(requestPtr)};
+    absStatus = graph.StartRun(std::move(sideInputPackets));
```
Comment (dkalinowski) [inline src/mediapipe_internal/mediapipegraphexecutor.cpp:767]: Is it really neccessary to move input packets? Mediapipe does not take copy of it, it takes reference

File: src/mediapipe_internal/mediapipegraphexecutor.cpp [hunk 3]
```diff
@@ -825,7 +829,12 @@ Status MediapipeGraphExecutor::infer(const KFSRequest* requestPtr, KFSResponse*
     }
     SPDLOG_DEBUG("Received all output stream packets for graph: {}", request->model_name());
     response->set_model_name(request->model_name());
+    // TODO we are not extracing timestamp here!
     response->set_id(request->id());
+    auto reqId = stoi64(request->id());
+    if (reqId.has_value()) {
```
Comment (dkalinowski) [inline src/mediapipe_internal/mediapipegraphexecutor.cpp:835]: ?

File: src/mediapipe_internal/mediapipegraphexecutor.cpp [hunk 4]
```diff
@@ -872,28 +881,40 @@ Status MediapipeGraphExecutor::infer(const KFSRequest* requestPtr, KFSResponse*
...
-        return Status(StatusCode::MEDIAPIPE_INVALID_TIMESTAMP, currentStreamTimestamp.DebugString());
+    return StatusCode::OK;
+}
+
+static inline Status checkTimestamp(const KFSRequest& request, const Timestamp& timestamp) {
+    if (!timestamp.IsRangeValue()) {
+        // TODO we could check it after automatic incrementing and disconnect from server side?
+        // to avoid unnecessary processing of next request. We would have to return information
+        // that it's last response
+        SPDLOG_DEBUG("Timestamp not in range: {}; for request to: ", timestamp.DebugString(), request.model_name());
```
Comment (dkalinowski) [inline src/mediapipe_internal/mediapipegraphexecutor.cpp:906]: Missing `{}`

File: src/mediapipe_internal/mediapipegraphexecutor.cpp [hunk 5]
```diff
@@ -929,6 +950,7 @@ Status MediapipeGraphExecutor::inferStream(const KFSRequest& firstRequest, ::grp
                     ::inference::ModelStreamInferResponse resp;
                     OVMS_RETURN_MP_ERROR_ON_FAIL(serializePacket(outputName, *resp.mutable_infer_response(), packet), "error in serialization");
                     resp.mutable_infer_response()->set_id(std::to_string(packet.Timestamp().Value()));
```
Comment (dkalinowski) [inline src/mediapipe_internal/mediapipegraphexecutor.cpp:952]: Do we still need set_id?

File: src/test/streaming_test.cpp [hunk 6]
```diff
@@ -47,13 +48,31 @@ class StreamingTest : public Test {
     MockedServerReaderWriter<::inference::ModelStreamInferResponse, ::inference::ModelInferRequest> stream;
 };
 
+static void setRequestTimestamp(KFSRequest& request, const std::string& value) {
+    request.clear_parameters();
+    SPDLOG_ERROR("requesting :{}", value);
```
Comment (dkalinowski) [inline src/test/streaming_test.cpp:53]: Remove ER and debug logs

File: src/test/streaming_test.cpp [hunk 7]
```diff
@@ -157,7 +177,8 @@ static auto ReceiveInvalidWithTimestampWhenNotified(std::vector<std::string> inp
     return [inputs, timestamp, &mtx](::inference::ModelInferRequest* req) {
         std::lock_guard<std::mutex> lock(mtx);
         prepareInvalidRequest(*req, inputs);
-        req->set_id(std::to_string(timestamp));
+        setRequestTimestamp(*req, std::to_string(timestamp));
+        //       req->set_id(std::to_string(timestamp));
```
Comment (dkalinowski) [inline src/test/streaming_test.cpp:181]: remove comments

**Review by dtrawins — APPROVED** [review]

**Pull request merged**